.github/workflows/ci.yml

```diff
@@ -622,8 +700,7 @@ jobs:
             --env "IREE_IMPORT_TFLITE_BIN=${TF_BINARIES_DIR}/iree-import-tflite" \
             --env "LLVM_BIN_DIR=${BUILD_DIR}/third_party/llvm-project/llvm/bin" \
             gcr.io/iree-oss/riscv@sha256:720bc0215d8462ea14352edc22710a6ce4c0c1daff581d179dd173885f1d8a35 \
-            bash -euo pipefail -c \
-              "./build_tools/cmake/build_riscv.sh && ./tests/riscv64/smoke.sh"
+            bash -euo pipefail -c "./tests/riscv64/smoke.sh"
```

Comment:
(I'll be adding web browser/platform tests here soon, refreshed from https://github.com/iree-org/iree/blob/main/build_tools/buildkite/samples.yml soon, so this is on my mind)

Separate from particular refactoring, I think we could adjust how we test platforms like riscv32/64 so we can make the most of our existing test suite.

Ideally we'd just have a list of supported (and possibly experimental) platforms, then the CI would build runtime targets, compile using some configuration, and execute the same test suites across that platform matrix. Each platform can also individually have tests for _platform specific_ behavior - what's in [smoke.sh](https://github.com/iree-org/iree/blob/main/tests/riscv64/smoke.sh) is not that.

See also test suites like https://github.com/iree-org/iree/blob/a456db644d1ae70ca3b89d3f478f09ec25d0a213/tests/e2e/xla_ops/BUILD#L393-L402

Reply:
This sort of relates to this particular PR, because it could be nice to just set a `matrix` in the yaml file (https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) for each `build_[platform]` and `test_[platform]`, without needing to copy/paste all this boilerplate.

Reply:
Tried a little bit on the matrix. Since there are a few parameters (architecture, docker image) needed to build each platform, it's not very clean to use the matrix to define them (instead of just a string for each platform, I use an object).

However, this reduces lots of boilerplate jobs, so I changed the PR to use the matrix. The thing I'm not satisfied is that right now we have the docker image SHA hash in the matrix, which is verbose and the UI will show that long hash:
![image](https://user-images.githubusercontent.com/2104162/187549434-2c339381-c384-44ac-8979-8bdc83bc664b.png)

For the next step, I feel like we can move the hash into `build_tools/github_actions/docker_run.sh`, so we only need to specify the platform string (e.g. riscv).

I also standardize the `build_riscv.sh` a little, which should enable us to use the matrix for the test builds. Same change also needs to be done in `build_android.sh`.

Reply:
> Tried a little bit on the matrix. Since there are a few parameters (architecture, docker image) needed to build each platform, it's not very clean to use the matrix to define them (instead of just a string for each platform, I use an object).

Yeah I had some ideas for that, like the CI pipeline just running an entry point script that would then branch based on platform (`build_platform.sh` -> `build_riscv64.sh` / `build_emscripten.sh`, same for `test_platform.sh`). Each platform may use a different build system (as is the case with Emscripten for the web - it has a CMake wrapper for builds, then tests will need to run under node or in browsers... might not be easy under `ctest`).

Reply:
Yeah I agree more standardization would help here.